paperqa/llms.py

```diff
@@ -398,6 +427,108 @@ async def similarity_search(
+        batch_size: int = 100,
+        max_concurrent_requests: int = 5,
+    ) -> Docs:
+        """
+        Reconstruct a Docs object from a Qdrant collection.
+
+        Args:
+            client: AsyncQdrantClient instance
+            collection_name: Name of the collection to load
+            vector_name: Optional name of the vector field if using named vectors
+            batch_size: Number of records to fetch per batch
+            max_concurrent_requests: Maximum number of concurrent scroll requests
+
+        Returns:
+            Reconstructed Docs object
+        """
+        from paperqa.docs import Docs
+
+        vectorstore = cls(
+            client=client, collection_name=collection_name, vector_name=vector_name
+        )
+        docs = Docs(texts_index=vectorstore)
+
+        # Get collection info to determine total points
+        collection_info = await client.get_collection(collection_name)
+        total_points = collection_info.points_count
+
+        async def fetch_batch(offset: int) -> list:
+            return (
+                await client.scroll(
+                    collection_name=collection_name,
+                    limit=batch_size,
+                    offset=offset,
+                    with_payload=True,
+                    with_vectors=True,
+                )
+            )[0]
+
+        # Create a queue of offsets to process
+        offset_queue = deque(range(0, total_points, batch_size))
+        active_tasks = set()
+        all_points = []
+
+        # Process batches concurrently
+        while offset_queue or active_tasks:
+            # Start new tasks if we have capacity and offsets to process
+            while len(active_tasks) < max_concurrent_requests and offset_queue:
+                offset = offset_queue.popleft()
+                task = asyncio.create_task(fetch_batch(offset))
+                active_tasks.add(task)
+
+            if not active_tasks:
+                break
+
+            # Wait for at least one task to complete
+            done, active_tasks = await asyncio.wait(
+                active_tasks, return_when=asyncio.FIRST_COMPLETED
+            )
+
+            # Process completed tasks
+            for task in done:
+                points = await task
+                all_points.extend(points)
+
+        # Process all points
+        for point in all_points:
+            try:
+                payload = point.payload
+                doc_data = payload["doc"]
+
+                # Only create new Doc if we haven't seen this dockey
+                if doc_data["dockey"] not in docs.docs:
+                    docs.docs[doc_data["dockey"]] = Doc(
+                        docname=doc_data["docname"],
+                        citation=doc_data["citation"],
+                        dockey=doc_data["dockey"],
+                    )
+                    docs.docnames.add(doc_data["docname"])
+
+                # Create Text object
+                text = Text(
```

Comment:
```suggestion
                text = Text(
```

This comment is redundant to the code